### PR TITLE
Use memcache for new index response

### PR DESCRIPTION
**Pull request by sonalkr132**
I am not sure if this is what was needed in #141 Still, here goes nothing

### CODE REVIEW

File: lib/bundler_api/web.rb [hunk 1]
```diff
@@ -273,6 +273,24 @@ def get_cached_dependencies
...
+    @dalli_client.get(key) || begin
+      calc_info = @gem_info.info(name)
+      @dalli_client.set(key, calc_info)
+      calc_info
+    end
+  end
+
+  def get_cached_names
+    key = "names"
+    @dalli_client.get(key) || begin
```
Comment (segiddins) [inline lib/bundler_api/web.rb]: this should use `fetch` passing in a block that'll get called if the key isn't set

**Comment by sonalkr132** [comment]
@segiddins I have made the suggested change.

When I tried to cache `/versions` ([this](https://github.com/bundler/bundler-api/blob/master/lib/bundler_api/web.rb#L157-L164)) I got following error:
> Value for versions over max size: 1048576 <= 9713704

What should I do here?
related info: [What is the maximum size of an object you can store in memcached? Is that configurable?](http://docs.oracle.com/cd/E17952_01/refman-5.6-en/ha-memcached-faq.html#qandaitem-17-6-5-1-2)

File: lib/bundler_api/web.rb [hunk 2]
```diff
@@ -273,6 +273,24 @@ def get_cached_dependencies
     dependencies
   end
 
+  def get_cached_info(name)
+    key = "info/#{name}"
+    @dalli_client.fetch(key) do
+      calc_info = @gem_info.info(name)
+      @dalli_client.set(key, calc_info)
```
Comment (segiddins) [inline lib/bundler_api/web.rb]: setting it manually is unnecessary (http://www.rubydoc.info/gems/dalli/2.7.6/Dalli%2FClient%3Afetch)

Comment (sonalkr132) [inline lib/bundler_api/web.rb]: Thanks. I am wondering if I don't need a new method just for two lines:
```Ruby
key = "info/#{name}"
@dalli_client.fetch(key) { @gem_info.info(name) }
```
I can just write:
```Ruby
  get "/info/:name" do
    status 404 unless NEW_INDEX_ENABLED
    etag_response_for(params[:name]) do
      @dalli_client.fetch("info/#{name}") { @gem_info.info(name) }
    end
  end
```
What do you think?

Comment (segiddins) [inline lib/bundler_api/web.rb]: :+1: looks good

**Comment by segiddins** [comment]
@indirect / @arthurnn r?

**Comment by indirect** [comment]
oops @homu r+

**Comment by indirect** [comment]
@sonalkr132 oh this needs a rebase I think

**Comment by sonalkr132** [comment]
@indirect It is merged https://github.com/bundler/bundler-api/commit/dfa69fb8d30b8460fb4b56572eaa752f05ea30b4

**Comment by indirect** [comment]
@sonalkr132 I guess my web view merge button was just broken 😓 thanks!

**Pull request merged**